packages/commerce-sdk-react/src/hooks/useAsync.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {ActionResponse, QueryResponse} from './types'
+
+export const useAsync = <T>(fn: () => Promise<T>, deps?: unknown[]): QueryResponse<T> => {
```

Comment:
I imagine this abstraction will be replaced by `useServerEffect` (server side) and `useSWR` (client side).